lab_1/main.py

```diff
@@ -4,56 +4,168 @@
+        sorted_dict = {k: freq_dict[k] for k in sorted(freq_dict, key=freq_dict.get, reverse=True)}
+        for key in sorted_dict:
+            top_words.append (key)
+        return top_words[:top_n]
+top_words = get_top_n_words(freq_dict, 2)
+'''        for key, value in sorted (freq_dict.items(), key = freq_dict.get, reverse = True):
+            top_words.append (key)
+        return top_words[:top_n-1]
+'''
+'''         if top_words == []:
+                top_words.append (key)
+            else:
+                for word in top_words:
+                    if value > freq_dict[word]:
+                        top_words = [key] + top_words
+                    else:
+                        top_words.append (key)
+    return top_words[:top_n-1]'''
+"""
     Returns the most common words
     :param freq_dict: a dictionary with frequencies
     :param top_n: a number of the most common words to return
     :return: a list of the most common words
     e.g. tokens = ['weather', 'sunny', 'man', 'happy', 'and', 'dog', 'happy']
     top_n = 1
     --> ['happy']
-    """
-    pass
+"""
 
 
-def get_concordance(tokens: list, word: str, left_context_size: int, right_context_size: int) -> list:
-    """
+def get_concordance(tokens, word, left_context_size, right_context_size):
+    concordance = []
+    if type(tokens) != list or type(word) != str: # or word not in tokens
+        return []
+    elif type(left_context_size) != int or type(right_context_size) != int:
+        return []
+    elif type(left_context_size) == int and type(right_context_size) == int and left_context_size < 1 and right_context_size < 1:
+        return []
+    elif right_context_size < 1 and left_context_size >= 1:
+        for number, token in enumerate (tokens):
+            context = []
+            if token == word:
+                left_size = left_context_size
+                while left_size != 0:
+                    if number - left_size >= 0:
+                        context.append (tokens[number-left_size])
+                    left_size -= 1
+                context.append (token)
+            if context != []:
+                concordance.append (context)
+    elif right_context_size >= 1 and left_context_size < 1:
+        for number, token in enumerate (tokens):
+            context = []
+            if token == word:
+                context = [token]
+                iterations = 1
+                right_size = right_context_size
+                while right_size != 0:
+                    if len(tokens) > number + iterations:
+                        context.append (tokens[number+iterations])
+                    right_size -= 1
+                    iterations += 1
+            if context != []:
+                concordance.append (context)
+    elif right_context_size >= 1 and left_context_size >= 1:
+        for number, token in enumerate (tokens):
```

Comment:
almost the same code here and in the next if-statement, think how you can write it once and reuse in different if-statements